torch/_functorch/aot_autograd.py

```diff
@@ -1137,20 +1137,22 @@ def dispatch_and_compile():
             )
         return compiled_fn
 
-    # Autograd cache stuff
-    remote = should_use_remote_autograd_cache()
-    local = should_use_local_autograd_cache()
     # We only care if the forward will return an OutputCode.
-    if (local or remote) and isinstance(fw_compiler, SerializableAOTDispatchCompiler):
-        compiled_fn = AOTAutogradCache.load(
-            dispatch_and_compile,
-            mod,
-            fake_flat_args,
-            aot_config,
-            cudagraphs,
-            local,
-            remote,
-        )
+    if isinstance(fw_compiler, SerializableAOTDispatchCompiler):
+        local = should_use_local_autograd_cache()
```

Comment:
This makes it so that we don't spend time calling these `should_use_local_autograd_cache` functions if there's no way we're caching anyway